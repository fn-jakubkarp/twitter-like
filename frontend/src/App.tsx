import { FC } from 'react';

import { Box, Paper, Typography } from '@mui/material';

import './styles/main.scss';

const App: FC = () => {
  return (
    <Box className="center">
      <Paper sx={{ padding: 4 }}>
        <Typography variant="h6">Hello, please start here. ğŸ™‚</Typography>
      </Paper>
    </Box>
  );
};

export default App;
