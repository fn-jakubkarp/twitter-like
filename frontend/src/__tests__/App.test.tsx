import renderer from 'react-test-renderer';

import App from '@/src/App';
import '@testing-library/jest-dom';

describe('App', () => {
  it('should match snapshot', () => {
    const snapshot = renderer.create(<App />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
