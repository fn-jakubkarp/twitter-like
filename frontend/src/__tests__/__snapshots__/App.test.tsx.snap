// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; min-height: 100vh;"
  >
    <div
      style="flex-grow: 1; display: flex; flex-direction: column;"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-4issou-MuiPaper-root-MuiAppBar-root"
        data-testid="header"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yurbhn-MuiSvgIcon-root"
            data-testid="TwitterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
          <div
            class="MuiTypography-root MuiTypography-h6 css-xq2o8l-MuiTypography-root"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1snpnll-MuiButtonBase-root-MuiButton-root"
              href="/"
              tabindex="0"
            >
              Twitter
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1snpnll-MuiButtonBase-root-MuiButton-root"
              data-testid="header-btn-home"
              href="/"
              tabindex="0"
            >
              Home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        style="flex-grow: 1; padding: 24px;"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
          data-testid="homepage-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1dka0wx-MuiPaper-root"
                data-testid="homepage-paper"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1syvpm1-MuiTypography-root"
                  data-testid="homepage-title"
                >
                  Home
                </h2>
                <form
                  class="MuiBox-root css-178yklu"
                  data-testid="createpostform-container"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-1fa0726-MuiFormControl-root-MuiTextField-root"
                    data-testid="createpostform-author-input"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14usnb2-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for=":r0:"
                      id=":r0:-label"
                    >
                      Author
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-u5cgm3-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r0:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Author
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-1fa0726-MuiFormControl-root-MuiTextField-root"
                    data-testid="createpostform-body-input"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14usnb2-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for=":r1:"
                      id=":r1:-label"
                    >
                      Post Content
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-5ba852-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r1:"
                        rows="4"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Post Content
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ghvi08-MuiButtonBase-root-MuiButton-root"
                    data-testid="createpostform-submit-btn"
                    tabindex="0"
                    type="submit"
                  >
                    Create Post
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-1o5hxn5-MuiDivider-root"
                data-testid="homepage-divider"
              />
              <div>
                Loading...
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
